### ISSUE TITLE
[ERROR] Unable to connect to the dependency-check database

### ISSUE THREAD

**Issue by brileyd** [dependency-check/DependencyCheck#2804]
I just upgraded dependency-check from 5.3.2 to 6.0.0 using brew.  Afterwards I get the error from the bug title using the cli.

**The errors in the log file are:**
2020-09-09 12:46:25,262 org.owasp.dependencycheck.data.nvdcve.ConnectionFactory:191
DEBUG -
org.owasp.dependencycheck.data.nvdcve.DatabaseException: Upgrade SQL file does not exist: data/upgrade_4.2.sql
        at org.owasp.dependencycheck.data.nvdcve.ConnectionFactory.updateSchema(ConnectionFactory.java:371)
        at org.owasp.dependencycheck.data.nvdcve.ConnectionFactory.ensureSchemaVersion(ConnectionFactory.java:423)
        at org.owasp.dependencycheck.data.nvdcve.ConnectionFactory.initialize(ConnectionFactory.java:189)
        at org.owasp.dependencycheck.data.nvdcve.ConnectionFactory.getConnection(ConnectionFactory.java:230)
        at org.owasp.dependencycheck.data.nvdcve.CveDB.open(CveDB.java:266)
        at org.owasp.dependencycheck.data.nvdcve.CveDB.<init>(CveDB.java:236)
        at org.owasp.dependencycheck.Engine.openDatabase(Engine.java:1047)
        at org.owasp.dependencycheck.Engine.doUpdates(Engine.java:923)
        at org.owasp.dependencycheck.Engine.initializeAndUpdateDatabase(Engine.java:740)
        at org.owasp.dependencycheck.Engine.analyzeDependencies(Engine.java:670)
        at org.owasp.dependencycheck.App.runScan(App.java:254)
        at org.owasp.dependencycheck.App.run(App.java:186)
        at org.owasp.dependencycheck.App.main(App.java:81)
Caused by: java.lang.IllegalArgumentException: resource data/upgrade_4.2.sql not found.
        at com.google.common.base.Preconditions.checkArgument(Preconditions.java:217)
        at com.google.common.io.Resources.getResource(Resources.java:195)
        at org.owasp.dependencycheck.data.nvdcve.ConnectionFactory.updateSchema(ConnectionFactory.java:358)
        ... 12 common frames omitted

**And:**
2020-09-09 12:46:27,959 org.owasp.dependencycheck.App:209
DEBUG - unexpected error
org.owasp.dependencycheck.data.nvdcve.DatabaseException: Unable to connect to the dependency-check database
        at org.owasp.dependencycheck.Engine.throwFatalDatabaseException(Engine.java:780)
        at org.owasp.dependencycheck.Engine.initializeAndUpdateDatabase(Engine.java:747)
        at org.owasp.dependencycheck.Engine.analyzeDependencies(Engine.java:670)
        at org.owasp.dependencycheck.App.runScan(App.java:254)
        at org.owasp.dependencycheck.App.run(App.java:186)
        at org.owasp.dependencycheck.App.main(App.java:81)
Caused by: org.owasp.dependencycheck.data.nvdcve.DatabaseException: Database schema does not match this version of dependency-check
        at org.owasp.dependencycheck.data.nvdcve.ConnectionFactory.initialize(ConnectionFactory.java:192)
        at org.owasp.dependencycheck.data.nvdcve.ConnectionFactory.getConnection(ConnectionFactory.java:230)
        at org.owasp.dependencycheck.data.nvdcve.CveDB.open(CveDB.java:266)
        at org.owasp.dependencycheck.data.nvdcve.CveDB.<init>(CveDB.java:236)
        at org.owasp.dependencycheck.Engine.openDatabase(Engine.java:1047)
        at org.owasp.dependencycheck.Engine.doUpdates(Engine.java:923)
        at org.owasp.dependencycheck.Engine.initializeAndUpdateDatabase(Engine.java:740)
        ... 4 common frames omitted
Caused by: org.owasp.dependencycheck.data.nvdcve.DatabaseException: Upgrade SQL file does not exist: data/upgrade_4.2.sql
        at org.owasp.dependencycheck.data.nvdcve.ConnectionFactory.updateSchema(ConnectionFactory.java:371)
        at org.owasp.dependencycheck.data.nvdcve.ConnectionFactory.ensureSchemaVersion(ConnectionFactory.java:423)
        at org.owasp.dependencycheck.data.nvdcve.ConnectionFactory.initialize(ConnectionFactory.java:189)
        ... 10 common frames omitted
Caused by: java.lang.IllegalArgumentException: resource data/upgrade_4.2.sql not found.
        at com.google.common.base.Preconditions.checkArgument(Preconditions.java:217)
        at com.google.common.io.Resources.getResource(Resources.java:195)
        at org.owasp.dependencycheck.data.nvdcve.ConnectionFactory.updateSchema(ConnectionFactory.java:358)
        ... 12 common frames omitted
2020-09-09 12:46:27,959 org.owasp.dependencycheck.App:82
DEBUG - Exit code: -13

I was able to resolve the issue by deleting the contents of /usr/local/var/dependencycheck/ which meant I had to redownload all definitions.

**Comment by mvits** [comment]
Same problem, solution for the moment is to delete data directory of before version.

**Comment by petergphillips** [comment]
In case anyone has trouble finding the data directory, it is located at `~/.gradle/dependency-check-data` so 
```
rm -r ~/.gradle/dependency-check-data
```
worked for me.

**Comment by karuppiah7890** [comment]
Same problem here. I'm going to apply the workaround, but what's the ideal solution here? I can help with a PR if possible 😄

**Comment by jeremylong** [comment]
@petergphillips you can also just run `./gradlew dependencyCheckPurge`.  

That being said - can you run `ls ~/.gradle/dependency-check-data` and post the results to show what is in the directory?  There should be a 5.0 directory. With earlier versions there should have been a 4.0 directory.

**Comment by petergphillips** [comment]
I just have a `4.0` directory I'm afraid, no matter whether I run 5.3.2.1 or 6.0.0. No other directories there. I'm running using Gradle 6.6.1 on ubuntu.  Tried without the gradle daemon but didn't make any difference.

**Comment by jeremylong** [comment]
Then something very odd is going on...  Something very strange is going on - the plugin should be using a 5.0 data directory:

https://github.com/jeremylong/dependency-check-gradle/blob/0c560f74d4b6c358d6722f915bf23c4f7153c833/src/main/resources/task.properties#L2

Any chance that somehow the 5.x plugin is still on the path somehow and the wrong task.properties is being picked up?

**Comment by petergphillips** [comment]
Isn't the value defined in https://github.com/jeremylong/dependency-check-gradle/blob/master/src/main/groovy/org/owasp/dependencycheck/gradle/extension/DataExtension.groovy#L29 instead?

**Comment by petergphillips** [comment]
I've raised https://github.com/jeremylong/dependency-check-gradle/pull/197 to hopefully fix

**Comment by jeremylong** [comment]
Sorry I missed the PR - I just implemented this and pushed 6.0.0.1.

Gradle users - please upgrade to 6.0.0.1.

**Comment by brileyd** [comment]
Does this correct the issue with the CLI version or just the gradle plugin as my issue was with the CLI version?

**Comment by jeremylong** [comment]
@brileyd there was a bug in the gradle plugin.  For the CLI if you run into the issue you just need to run `dependency-check.sh --purge` to remove the old database.

**Comment by jeremylong** [comment]
Warning messages have been updated in 6.0.2 for the CLI and updated paths in Ant and Gradle should resolve the actual bugs. However, 6.x did contain breaking changes and many may need to run `purge`.

**Comment by GeorgievGG** [comment]
Another reason that might cause the same error is lack of permission. I had my tool in the Program Files folder and permissions prevented the tool from making a DB. It was hard to realize what's going on until I ran purge and realized I have no rights to do it.

**Comment by leobut** [comment]
When I run `./gradlew dependencyCheckAggregate --stacktrace`, I get this error again. `./gradlew dependencyCheckPurge` does not help.

I am using Gradle 7.6 and Gradle Plugin version 8.1.0.

This is my configuration:
```
check.dependsOn dependencyCheckAggregate
dependencyCheck {
    format = 'ALL'
    scanProjects = [':customer:test']
    data {
        directory = './.ci/owasp-dependency-check/data'
    }
}
```

When I completely delete my data folder, the problem persists. The only "fix" I found so far was change the data folder to e.g. data2 at the end. There must be some database file outside my data folder that only gets rebuilt once I change the folder location.

Please advice.

Here is my stacktrace:
```
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':dependencyCheckAggregate'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:142)
	at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:282)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:140)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:128)
	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:77)
	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:57)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:69)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:322)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:309)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:302)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:288)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:462)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:379)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:49)
Caused by: org.gradle.api.GradleException: Analysis failed.
	at org.owasp.dependencycheck.gradle.tasks.AbstractAnalyze.analyze(AbstractAnalyze.groovy:104)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:125)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:58)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:51)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:29)
	at org.gradle.api.internal.tasks.execution.TaskExecution$3.run(TaskExecution.java:236)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:47)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:68)
	at org.gradle.api.internal.tasks.execution.TaskExecution.executeAction(TaskExecution.java:221)
	at org.gradle.api.internal.tasks.execution.TaskExecution.executeActions(TaskExecution.java:204)
	at org.gradle.api.internal.tasks.execution.TaskExecution.executeWithPreviousOutputFiles(TaskExecution.java:187)
	at org.gradle.api.internal.tasks.execution.TaskExecution.execute(TaskExecution.java:165)
	at org.gradle.internal.execution.steps.ExecuteStep.executeInternal(ExecuteStep.java:89)
	at org.gradle.internal.execution.steps.ExecuteStep.access$000(ExecuteStep.java:40)
	at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:53)
	at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:50)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
	at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:50)
	at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:40)
	at org.gradle.internal.execution.steps.RemovePreviousOutputsStep.execute(RemovePreviousOutputsStep.java:68)
	at org.gradle.internal.execution.steps.RemovePreviousOutputsStep.execute(RemovePreviousOutputsStep.java:38)
	at org.gradle.internal.execution.steps.CancelExecutionStep.execute(CancelExecutionStep.java:41)
	at org.gradle.internal.execution.steps.TimeoutStep.executeWithoutTimeout(TimeoutStep.java:74)
	at org.gradle.internal.execution.steps.TimeoutStep.execute(TimeoutStep.java:55)
	at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:51)
	at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:29)
	at org.gradle.internal.execution.steps.CaptureStateAfterExecutionStep.executeDelegateBroadcastingChanges(CaptureStateAfterExecutionStep.java:124)
	at org.gradle.internal.execution.steps.CaptureStateAfterExecutionStep.execute(CaptureStateAfterExecutionStep.java:80)
	at org.gradle.internal.execution.steps.CaptureStateAfterExecutionStep.execute(CaptureStateAfterExecutionStep.java:58)
	at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:48)
	at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:36)
	at org.gradle.internal.execution.steps.BuildCacheStep.executeWithoutCache(BuildCacheStep.java:181)
	at org.gradle.internal.execution.steps.BuildCacheStep.lambda$execute$1(BuildCacheStep.java:71)
	at org.gradle.internal.Either$Right.fold(Either.java:175)
	at org.gradle.internal.execution.caching.CachingState.fold(CachingState.java:59)
	at org.gradle.internal.execution.steps.BuildCacheStep.execute(BuildCacheStep.java:69)
	at org.gradle.internal.execution.steps.BuildCacheStep.execute(BuildCacheStep.java:47)
	at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:36)
	at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:25)
	at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:36)
	at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:22)
	at org.gradle.internal.execution.steps.SkipUpToDateStep.executeBecause(SkipUpToDateStep.java:110)
	at org.gradle.internal.execution.steps.SkipUpToDateStep.lambda$execute$2(SkipUpToDateStep.java:56)
	at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:56)
	at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:38)
	at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:73)
	at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:44)
	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:37)
	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:27)
	at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:89)
	at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:50)
	at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:102)
	at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:57)
	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:76)
	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:50)
	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.executeWithNoEmptySources(SkipEmptyWorkStep.java:254)
	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:91)
	at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:56)
	at org.gradle.internal.execution.steps.RemoveUntrackedExecutionStateStep.execute(RemoveUntrackedExecutionStateStep.java:32)
	at org.gradle.internal.execution.steps.RemoveUntrackedExecutionStateStep.execute(RemoveUntrackedExecutionStateStep.java:21)
	at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsStartedStep.execute(MarkSnapshottingInputsStartedStep.java:38)
	at org.gradle.internal.execution.steps.LoadPreviousExecutionStateStep.execute(LoadPreviousExecutionStateStep.java:43)
	at org.gradle.internal.execution.steps.LoadPreviousExecutionStateStep.execute(LoadPreviousExecutionStateStep.java:31)
	at org.gradle.internal.execution.steps.AssignWorkspaceStep.lambda$execute$0(AssignWorkspaceStep.java:40)
	at org.gradle.api.internal.tasks.execution.TaskExecution$4.withWorkspace(TaskExecution.java:281)
	at org.gradle.internal.execution.steps.AssignWorkspaceStep.execute(AssignWorkspaceStep.java:40)
	at org.gradle.internal.execution.steps.AssignWorkspaceStep.execute(AssignWorkspaceStep.java:30)
	at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:37)
	at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:27)
	at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:44)
	at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:33)
	at org.gradle.internal.execution.impl.DefaultExecutionEngine$1.execute(DefaultExecutionEngine.java:76)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:139)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:128)
	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:77)
	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:57)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:69)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:322)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:309)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:302)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:288)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:462)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:379)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:49)
Caused by: org.owasp.dependencycheck.exception.ExceptionCollection: One or more exceptions occurred during analysis:
	DatabaseException: Unable to connect to the dependency-check database
		caused by DatabaseException: Unable to create the database structure
		caused by DatabaseException: Unable to create database statement
		caused by JdbcSQLSyntaxErrorException: Cannot drop "VULNERABILITY" because "FKCWEENTRY" depends on it; SQL statement:
DROP TABLE IF EXISTS vulnerability [90107-214]
	at org.owasp.dependencycheck.Engine.throwFatalDatabaseException(Engine.java:746)
	at org.owasp.dependencycheck.Engine.initializeAndUpdateDatabase(Engine.java:712)
	at org.owasp.dependencycheck.Engine.analyzeDependencies(Engine.java:631)
	at org.owasp.dependencycheck.gradle.tasks.AbstractAnalyze.analyze(AbstractAnalyze.groovy:100)
	... 116 more
```

**Comment by nauni77** [comment]
Hello, I had the same problem. We have a HTTP/ HTTPS proxy, which needs to be used. Now we configured a proxy location for CVE data. But because of some issue the none proxy hosts are not working. That means, if I want to use the local CVE proxy I have to disable the HTTP/ HTTPS proxy at gradle settings. But because of this the other services can't be reached at the internet. During the build I see some timeouts but at the end it fails because of the exception:

```
Cannot drop "VULNERABILITY" because "FKCWEENTRY" depends on it; SQL statement:
DROP TABLE IF EXISTS vulnerability
```

After disabling all the other services and avoid the failing of these requests the database shows no problem anymore.

Here is my snipped of the configuration:
```
    dependencyCheck {
        failOnError = false
        autoUpdate = true
        cveValidForHours = 1
        format = 'ALL'
        failBuildOnCVSS = 11
        data {
            directory = rootProject.file('nvddata').absolutePath
        }
        cve {
            urlBase = 'https://repo.intra.mycompany.de/artifactory/remote-ext-http-owasp/feeds/json/cve/1.1/nvdcve-1.1-%d.json.gz'
            urlModified  = 'https://repo.intra.mycompany.de/artifactory/remote-ext-http-owasp/feeds/json/cve/1.1/nvdcve-1.1-modified.json.gz'
        }
        suppressionFile = 'owasp.suppression.xml'

        analyzers {
            knownExploitedEnabled = false
            ossIndex {
                enabled = false
	     }
	     retirejs {
                enabled = false
             }
             hostedSuppressions {
                 enabled = false
             }
        }
    }
```
The important part is the section of the analyzers.

In my eyes this is a bug and the settings from `.gradle/gradle.properties` should work correct and only the request to the cve files from `repo.intra.mycompany.de` should be transferred direct. But all other services (not named at the no_proxy variable) should be use the proxy.

Snippet from `~/.gradle/gradle.properties`:
```
systemProp.http.proxyHost=proxy.mycompany.de
systemProp.http.proxyPort=8080
systemProp.http.nonProxyHosts=localhost|127.0.0.1|10.10.1.*|192.168.*|10.254.236.*|*.intra.mycompany.de|repo.intra.mycompany.de

systemProp.https.proxyHost=proxy.mycompany.de
systemProp.https.proxyPort=8080
systemProp.https.nonProxyHosts=localhost|127.0.0.1|10.10.1.*|192.168.*|10.254.236.*|*.intra.mycompany.de|repo.intra.mycompany.de
```

The description about the proxy configuration: 
https://docs.gradle.org/current/userguide/build_environment.html#sec:accessing_the_web_via_a_proxy

Oli

**Comment by aroyarexs** [comment]
I run into this error as well.
In our case we set a custom `data.directory = 'myCustomDirectory/'`.
_dependencyCheckUpdate_ runs fine, but _dependencyCheckAnalyze_ then results in the above mentioned error.
I compared the content of `~/.gradle/dependency-check-data` with the one of the custom directory.
The difference was that in the default gradle directory OWASP folders per version are created containing the data like mentioned by @jeremylong.
Using the custom directory, no OWASP version folders are created.
Does the gradle plugin expects this version folders even when using a custom directory?

We used version 8.4.3.
